actors/builtin/power/power_test.go

```diff
@@ -752,6 +765,161 @@ func TestSubmitPoRepForBulkVerify(t *testing.T) {
+		sealInfo.SectorID = abi.SectorID{Number: abi.SectorNumber(i)}
+		return &sealInfo
+	}
+
+	t.Run("success with one miner and one confirmed sector", func(t *testing.T) {
+		rt, ac := basicPowerSetup(t)
+		info := sealInfo(1)
+		ac.submitPoRepForBulkVerify(rt, miner1, info)
+
+		infos := map[addr.Address][]abi.SealVerifyInfo{miner1: []abi.SealVerifyInfo{*info}}
+		cs := []confirmedSectorSend{{miner1, []int{1}}}
+
+		ac.onEpochTickEnd(rt, 0, big.Zero(), cs, infos)
+	})
+
+	t.Run("success with one miner and multiple confirmed sectors", func(t *testing.T) {
+		rt, ac := basicPowerSetup(t)
+		info1 := sealInfo(1)
+		info2 := sealInfo(2)
+		info3 := sealInfo(3)
+
+		ac.submitPoRepForBulkVerify(rt, miner1, info1)
+		ac.submitPoRepForBulkVerify(rt, miner1, info2)
+		ac.submitPoRepForBulkVerify(rt, miner1, info3)
+
+		infos := map[addr.Address][]abi.SealVerifyInfo{miner1: []abi.SealVerifyInfo{*info1, *info2, *info3}}
+		cs := []confirmedSectorSend{{miner1, []int{1, 2, 3}}}
+
+		ac.onEpochTickEnd(rt, 0, big.Zero(), cs, infos)
+	})
+
+	t.Run("success with multiple miners and multiple confirmed sectors and assert expected power", func(t *testing.T) {
+		owner := tutil.NewIDAddr(t, 100)
+		miner2 := tutil.NewIDAddr(t, 102)
+		miner3 := tutil.NewIDAddr(t, 103)
+		miner4 := tutil.NewIDAddr(t, 104)
+		powerUnit := power.ConsensusMinerMinPower
+
+		rt, ac := basicPowerSetup(t)
+		info1 := sealInfo(0)
+		info2 := sealInfo(1)
+		ac.submitPoRepForBulkVerify(rt, miner1, info1)
+		ac.submitPoRepForBulkVerify(rt, miner1, info2)
+
+		info3 := sealInfo(100)
+		info4 := sealInfo(101)
+		ac.submitPoRepForBulkVerify(rt, miner2, info3)
+		ac.submitPoRepForBulkVerify(rt, miner2, info4)
+
+		info5 := sealInfo(200)
+		info6 := sealInfo(201)
+		ac.submitPoRepForBulkVerify(rt, miner3, info5)
+		ac.submitPoRepForBulkVerify(rt, miner3, info6)
+
+		info7 := sealInfo(300)
+		info8 := sealInfo(301)
+		ac.submitPoRepForBulkVerify(rt, miner4, info7)
+		ac.submitPoRepForBulkVerify(rt, miner4, info8)
+
+		// TODO Because read order of keys in a multi-map is not as per insertion order,
+		// we have to move around the expected sends
+		cs := []confirmedSectorSend{{miner1, []int{0, 1}},
+			{miner3, []int{200, 201}},
+			{miner4, []int{300, 301}},
+			{miner2, []int{100, 101}}}
+
+		infos := map[addr.Address][]abi.SealVerifyInfo{miner1: []abi.SealVerifyInfo{*info1, *info2},
+			miner2: []abi.SealVerifyInfo{*info3, *info4},
+			miner3: []abi.SealVerifyInfo{*info5, *info6},
+			miner4: []abi.SealVerifyInfo{*info7, *info8}}
+
+		ac.createMinerBasic(rt, owner, owner, miner1)
+		ac.createMinerBasic(rt, owner, owner, miner2)
+		ac.createMinerBasic(rt, owner, owner, miner3)
+		ac.createMinerBasic(rt, owner, owner, miner4)
+		ac.updateClaimedPower(rt, miner1, powerUnit, powerUnit)
+		ac.updateClaimedPower(rt, miner1, powerUnit, powerUnit)
+		ac.updateClaimedPower(rt, miner1, powerUnit, powerUnit)
+		ac.updateClaimedPower(rt, miner1, powerUnit, powerUnit)
```

Comment:
I think this `UpdateNetworkKPI` call should be extracted to a different test from proof verifications.

Reply:
This is done.